src/pb_memory.cc

```diff
@@ -193,8 +223,72 @@ PbMemory::FillShmData(
         THROW_IF_CUDA_ERROR(cudaIpcGetMemHandle(
             reinterpret_cast<cudaIpcMemHandle_t*>(memory_data_shm), data));
       }
+#ifndef TRITON_PB_STUB
```

Comment:
I think PbMemory should be just responsible for transferring the memory to and from the stub process. It looks like PbMemory is currently doing more.

My mental model is that with this optimization if `cuda_pool_offset` is set we would create the pointer based on the start address of the memory pool and there shouldn't be any copies involved in the `PbMemory` class. Please feel free to correct my understanding.